### ISSUE TITLE
Comparison of traditional SAM and DNN classification

### ISSUE THREAD

**Issue by lewismc** [capstone-coal/pycoal_examples#1]
Hi @EvandroCT I revisited this example today and like it very much. I have a few questions though.

1. Were you ever able to determine the confidence/accuracy of the DNN pixel predictions Vs those created by SAM + USGS Spectral Library v6/7? I think this would be a useful exercise and it is certainly required for us to prove the usefulness of the generated classified GIS products.  
1. I think it is important for us to  mention, within this tutorial, exactly what Deep Learning Framework(s) is being used. I know we use tensorflow, keras and sklearn however without reading the source code is it not obvious what each of these is being used for. Also, I would like you to consider moving the https://github.com/EvandroCT/dl-generalize project into the Pycoal repository... unless you are using it elsewhere of course. Seeing as the code you put together here relies upon the training and model generation so much I would hate for that code to go missing one day. 
1. I would really like to think about how we can parallelize the pycoal classification process such that we improve runtime performance for classification tasks over large imagery typical to AVIRIS-C and AVIRIS-NG. Right now it takes way way way too long. You are well aware of this. I would like to get a pixel classification implementation running on AWS SageMaker on GPU's with the aim of us [improving pixel classification](https://github.com/capstone-coal/pycoal/blob/master/pycoal/mineral.py#L200-L222). There is a [body of documentation](https://github.com/aws/sagemaker-python-sdk/tree/master/src/sagemaker/tensorflow) on using TensorFlow with Keras in SageMaker. Please provide your comments. 

The last few months I have been working on automating the data acquisitions, and processing pipeline for pycoal and we nearly have an end-to-end system which will provide a rich Science Data System for COAL. Right now a huge bottle neck is the pixel classification so I need to address that.
